src/m3ninx/search/searcher/conjunction.go

```diff
@@ -98,3 +116,31 @@ func (s *conjunctionSearcher) Search(r index.Reader) (postings.List, error) {
 
 	return pl, nil
 }
+
+type byLengthAscending []postings.List
+
+func (l byLengthAscending) Len() int {
+	return len(l)
+}
+
+func (l byLengthAscending) Less(i, j int) bool {
+	return l[i].Len() < l[j].Len()
```

Comment:
I'm afraid `List.Len()` is not a constant time, might make sense to precompute them in the slice.

Reply:
https://github.com/m3db/m3/pull/3981/commits/09c3dc27f4e7b9dfad8e98027a7d32165d650abb